docs/examples/snek5000-tgv/src/snek5000_tgv/tgv.usr.f

```diff
@@ -92,22 +117,32 @@ subroutine userchk()
 c
 c      iostep_full = iostep
 c      call full_restart_save(iostep_full)
-
-      if (mod(istep,10).ne.0) return
-
-      sum_e1 = 0.
-      sum_e2 = 0.
-      call curl(omg,vx,vy,vz,.false.,w1,w2)
-      do i = 1,n
-         vv = vx(i,1,1,1)**2 + vy(i,1,1,1)**2 + vz(i,1,1,1)**2
-         oo = omg(i,1)**2 + omg(i,2)**2 + omg(i,3)**2
-         sum_e1 = sum_e1 + vv*bm1(i,1,1,1)
-         sum_e2 = sum_e2 + oo*bm1(i,1,1,1)
-      enddo
-      e1 = 0.5 * glsum(sum_e1,1) / volvm1
-      e2 = 0.5 * glsum(sum_e2,1) / volvm1
-      if (nid.eq.0) write(6,2) time, e1, e2
-  2                 format(1p3e13.4,' monitor')
+     
+      if ((floor(time/period_save)).gt.(floor(t_last_save/
+     &period_save))) then
+      
+         t_last_save = time
+      
+         sum_e1 = 0.
+         sum_e2 = 0.
+         call curl(omg,vx,vy,vz,.false.,w1,w2)
+         do i = 1,ntot
+            vv = vx(i,1,1,1)**2 + vy(i,1,1,1)**2 + vz(i,1,1,1)**2
+            oo = omg(i,1)**2 + omg(i,2)**2 + omg(i,3)**2
+            sum_e1 = sum_e1 + vv*bm1(i,1,1,1)
+            sum_e2 = sum_e2 + oo*bm1(i,1,1,1)
+         enddo
+         e1 = 0.5 * glsum(sum_e1,1) / volvm1
+         e2 = 0.5 * glsum(sum_e2,1) / volvm1
+         if (nid.eq.0) then
+            write(6,2) time, e1, e2
+  2         format(1p3e13.4,' monitor')
+  
+            open(10, File=file_name, position='append')
+            write(10,'(F22.15,A,F22.15,A,F22.15)') time,',',e1,',',e2
```

Comment:
Why `F22.15`. It might be more general to save in scientific format. In Python, I would use `{.8g}`. In Fortran, you should be able to use `g14.8` or maybe even `g.8` ?